### ISSUE TITLE
riot turret bug

### ISSUE THREAD

**Issue by Westbend** [CleverRaven/Cataclysm-DDA#58435]
### Describe the bug

While driving, riot turrets will shoot you through armored windows with beanbags for full damage/stun.

### Steps to reproduce

1. drive armored APC
2. suddenly run into riot turret barricade
3. not able to stop in time and get shot to shit with bean bags
4. if armored, survive and back up.
5. if unarmored, most likely get seriously wounded or die.

### Expected behavior

Expected behavior, laugh as the beanbags bounce off my armored windows.

### Versions and configuration

cdda-windows-tiles-x64-msvc-2022-06-11-0641

**Comment by sonphantrung** [comment]
Hey, I have a question: Where does it shoot from?

**Comment by sonphantrung** [comment]
Also though, have you thoroughly check the windows and see if it has any armor platings?

**Comment by Westbend** [comment]
1. The riot turret shoots me from the front, through the reinforced glass windows as I am traveling and running into them. Also shoots through heavy-duty doors from the side if I am to close when trying to go around. 

2. Does that matter? A reinforced APC window should be up to the task of halting a beanbag regardless of attached extra armor plating one would assume?

**Comment by sonphantrung** [comment]
Check ur windshield's health

**Comment by Westbend** [comment]
The last time it happened two days ago, humvee, lowest reinforced window health, yellow. No visible sprite cracks. Also shot through the side, through the heavy-duty door window while driving around it.

But this happens regardless window health. Ive had this happen to me ever since I started playing CDDA 5 months ago, regardless of experimental game version. I have had a huge armored car, mint new from my shop, green in everything, and ill still get shot through the reinforced window. One design that didnt give me this problem was when I removed all windows and just drove an armored box with cameras.

When I finish my newest armored conveyance I will take a screenshot next time I run into a riot turret.

**Comment by a-chancey** [comment]
Yeah it can shoot through armored windows and not damage them. Not sure on the logic there other than the fact that those beanbags are technically grenades fired from 6 grenade launchers (disassemble a riot turret), so I'd expect armored glass to maybe not protect you, but it should also break

**Comment by Westbend** [comment]
> Not sure on the logic there other than the fact that those beanbags are technically grenades fired from 6 grenade launchers (disassemble a riot turret), so I'd expect armored glass to maybe not protect you, but it should also break

But lets not forget these are Riot turrets. If you are firing bean bags strong enough, and at velocities hard enough to punch through military reinforced windows, its not gonna be remotely "non-lethal", heh.

**Comment by sonphantrung** [comment]
Hmm, according to [HHG](https://nornagon.github.io/cdda-guide/#/item/40x46mm_m1006), the ammo used by riot turrets deals 10 DAMAGE?!

**Comment by a-chancey** [comment]
I don’t think y’all are realizing what’s being shot at you here. 

The riot control platforms are an array of 6 [M203 Grenade Launchers](https://en.m.wikipedia.org/wiki/M203_grenade_launcher), like what can be attached to an M4 or M16. 

[This is an example of a 40x46mm sponge round](https://www.directgovsource.com/product/exact-impact-40-mm-sponge-round/). They are “less lethal”, ineffective over 50 meters, will knock the wind out of you between 10-50 meters (and possibly break bones). Below 10 meters, they’re far more likely to be lethal due to their high muzzle velocity. 

Sponge rounds specifically are a type of round with a large, 1oz hard solid foam tip and a plastic body behind them to help stabilize. 

Again, these are being fired from grenade launchers, the rounds have quite a bit of oomph. 

They kill IRL, and in game, they’re not going to kill if you’re not already injured and get unlucky enough to take a few to the face, but that stun (which is what they are designed to do) is brutal.

**Comment by Westbend** [comment]
> [This is an example of a 40x46mm sponge round](https://www.directgovsource.com/product/exact-impact-40-mm-sponge-round/). They are “less lethal”, ineffective over 50 meters, will knock the wind out of you between 10-50 meters (and possibly break bones). Below 10 meters, they’re far more likely to be lethal due to their high muzzle velocity.

Yes, but we are not discussing their lethality, we are discussing the ability of a 1oz hard solid foam tip and a plastic body projectile being able to phase through military reinforced windshields of APCs without leaving any damage or meeting any resistance from said reinforced window. 

Maybe if I was driving an old Ford Fiesta, the windshield would get cracked, maybe even busted, but not even then I believe it would get through on the first shot. Especially without leaving any damage.

**Comment by a-chancey** [comment]
I agree there. Either the glass should be damaged or it should block the shot. It's a larger than 1" diameter hardened foam dart. It should be punching a hole in the windshield if it's hitting you

**Comment by sonphantrung** [comment]
I'm very aware of the lethality of the beanbag ammo, and the fact that 10 damage can go through a reinforced windshield seems weird indeed.

**Comment by a-chancey** [comment]
> @a-chancey Maybe the riot turrets need a description update then?

The descriptions point out that they’re *supposed* to be non lethal but police and prison guards showed that to not always be true. Sponge rounds themselves also say that they can be lethal in game, at least based on the HHG item description

**Comment by anothersimulacrum** [comment]
The beanbag should get the `NO_PENETRATE_OBSTACLES` ammo effect.
